src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,311 @@
+            String userInputLowercase = userInput.toLowerCase();
+
+            if (checkFeeding(userInputLowercase)) {
+                ui.happy();
+                continue;
+            } else if (checkIfBaseCommand(userInputLowercase)) {
+                handleBaseCommand(userInput.split(" "));
+                storage.writeToFile(tasklist);
+            } else if (checkIfFind(userInputLowercase)) {
+                handleFind(userInput.split(" "));
+                storage.writeToFile(tasklist);
+            } else if (checkIfLeave(userInputLowercase)) {
+                ui.goodbye();
+                break;
+            } else if (checkIfList(userInputLowercase)) {
+                handleList(tasklist);
+            } else if (checkIfTodo(userInputLowercase)) {
+                handleTodo(userInput, tasklist);
+                storage.writeToFile(tasklist);
+            } else if (checkIfEvent(userInputLowercase)) {
+                handleEvent(userInput, tasklist);
+                storage.writeToFile(tasklist);
+            } else if (checkIfDeadline(userInputLowercase)) {
+                handleDeadline(userInput, tasklist);
+                storage.writeToFile(tasklist);
+            } else {
+                ui.instructionMessage();
+            }
+        }
+    }
+
+    /**
+     * Handles the user input command for adding an event task.
+     * Parses the input string, creates an event task, and adds it to the task list.
+     *
+     * @param s The user input command string for adding an event task.
+     * @param t The task list to which the event task will be added.
+     */
+    public void handleEvent(String s, TaskList t) {
+        String eventName = "";
+        String[] temp = s.split(" ");
+        if (temp.length == 1 || temp[1].startsWith("/from")) {
+            System.out.println("Duke.Duke.Event cannot be blank");
+            return;
+        }
+        for (int a = 1; a < temp.length; a++) {
+            if (temp[a].startsWith("/from")) {
+                break;
+            }
+            eventName = eventName.concat(temp[a]);
+            eventName = eventName.concat(" ");
+        }
+        try {
+            String[] findPeriod = s.split(" /from ");
+            String start = findPeriod[1].split(" /to ")[0];
+            String end = findPeriod[1].split(" /to ")[1];
+            if (!canBeHandled(start) || !canBeHandled(end)) {
+                System.out.println("Please enter a event with the format event eventName /from dd/mm/yyyy /to dd/mm/yyyy!");
+                return;
+            }
+            Task ne = new Event(eventName, DateConvert(start), DateConvert(end));
+            t.add(ne);
+            System.out.println("Duke.Task added! You now have " + t.length() + " tasks to attend to.");
+            return;
+
+        } catch (ArrayIndexOutOfBoundsException b) {
+            System.out.println("Please enter a event with the format event eventName /from dd/mm/yyyy /to dd/mm/yyyy!");
```

Comment:
Abstract out the ui to show the error rather than hardcoding it in the method
```suggestion
         ui.showError("Event cannot be blank");

```